DAQController.cc

```diff
@@ -243,12 +245,22 @@ void DAQController::ReadData(int link){
     if (local_buffer.size() > 0) {
       fDataRate += local_size;
       int selector = (fCounter++)%fNProcessingThreads;
+      auto t_start = std::chrono::high_resolution_clock::now();
       fFormatters[selector]->ReceiveDatapackets(local_buffer, local_size);
+      auto t_end = std::chrono::high_resolution_clock::now();
+      mutex_wait_times.push_back(std::chrono::duration_cast<std::chrono::nanoseconds>(
+            t_end-t_start).count());
       local_size = 0;
     }
     readcycler++;
     std::this_thread::sleep_for(sleep_time);
   } // while run
+  if (mutex_wait_times.size() > 0) {
+    std::ofstream fout("/live_data/test/mutex_"+fHostname+"_ro"+std::to_string(link),
+        std::ios::binary);
+    fout.write((char*)mutex_wait_times.data(), sizeof(mutex_wait_times[0])*mutex_wait_times.size());
+    fout.close();
+  }
```

Comment:
This comment mildly confused me because it's referencing an old commit.
- `mutex_wait_times` is a vector, we track _all_ the values
- Originally these data were dumped to disk, but this is harder to work with than just outputting the interesting numbers to the logs.

Reply:
You answered my question with the new commit :)